subprojects/core-api/src/main/java/org/gradle/api/artifacts/dsl/DependencyAdder.java

```diff
@@ -85,48 +92,54 @@ public interface DependencyAdder {
      * @param version the version
      * @param classifier the classifier
      * @param extension the extension
+     * @see DependencyFactory#create(String, String, String, String, String)
      */
     void add(@Nullable String group, String name, @Nullable String version, @Nullable String classifier, @Nullable String extension);
 
     /**
-     * Add a dependency.
+     * Add a dependency and configure it.
      *
      * @param group the group
      * @param name the name
      * @param version the version
      * @param classifier the classifier
      * @param extension the extension
      * @param configuration an action to configure the dependency
+     * @see DependencyFactory#create(String, String, String, String, String)
      */
     void add(@Nullable String group, String name, @Nullable String version, @Nullable String classifier, @Nullable String extension, Action<? super ExternalModuleDependency> configuration);
 
     /**
      * Add a dependency.
      *
      * @param project project to add as a dependency
+     * @see DependencyFactory#create(Project)
      */
     void add(Project project);
```

Comment:
Reminder to remove these here and `DependencyFactory`.

Reply:
We need DependencyFactory to be able to go from Project -> ProjectDependency